sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -0,0 +1,1660 @@
+   graph.
+5. Support the USM model of memory as well as buffer/accessor model.
+6. Compatible with other SYCL extensions and features, e.g. kernel fusion &
+   built-in kernels.
+7. Ability to record a graph with commands submitted to different devices in the
+   same context.
+8. Capability to serialize graphs to a binary format which can then be
+   de-serialized and executed. This is helpful for AOT cases where a graph
+   can be created by an offline tool to be loaded and run without the end-user
+   incurring the overheads of graph creation.
+9. Backend interoperability, the ability to retrieve a native graph object from
+    the graph and use that in a native backend API.
+
+To allow for prototype implementations of this extension to be developed
+quickly for evaluation the scope of this proposal was limited to a subset
+of these requirements. In particular, the serialization functionality (8),
+backend interoperability (9), and a profiling/debugging interface (3) were
+omitted. As these are not easy to abstract over several backends without
+significant investigation. It is also hoped these features can be exposed as
+additive changes to the API, and thus introduced in future versions of the
+extension.
+
+Another reason for deferring a serialize/deserialize API (8) is that its scope
+could extend from emitting the graph in a binary format, to emitting a
+standardized IR format that enables further device specific graph optimizations.
+
+Multi-device support (7) is something that we are considering introducing into
+the extension in later revisions, which may result in API changes. It has been
+planned for to the extent that the definition of a graph node is device
+specific, however currently all nodes in a graph must target the same device
+provided to the graph constructor.
+
+=== Graph Building Mechanisms
+
+This extension contains two different API mechanisms for constructing a graph
+of commands:
+
+1. **Explicit graph building API** - Allows users to specify the exact nodes
+and edges they want to add to the graph.
+
+2. **Queue recording API (aka "Record & Replay")** - Introduces state to a
+`sycl::queue` such that rather than scheduling commands immediately for
+execution, they are added to the graph object instead, with edges captured from
+the dependencies of the command group.
+
+Each of these mechanisms for constructing a graph have their own advantages, so
+having both APIs available allows the user to pick the one which is most
+suitable for them. The queue recording API allows quicker porting of existing
+applications, and can capture external work that is submitted to a queue, for
+example via library function calls. While the explicit API can better express
+what data is internal to the graph for optimization, and dependencies don't need
+to be inferred.
+
+It is valid to combine these two mechanisms, however it is invalid to modify
+a graph using the explicit API while that graph is currently being recorded to,
```

Comment:
is it "... being recorded too". instead of ".. to" ?

Reply:
I think that "to" rather than "too" is correct, but rephrased this wording to "it is invalid to modify a graph using the explicit API while that graph is currently recording commands" to avoid any confusion over grammar.